packages/features/schedules/lib/use-schedule/useNonEmptyScheduleDays.ts

```diff
@@ -3,7 +3,7 @@ import { useMemo } from "react";
 import type { Slots } from "../use-schedule";
 
 export const getNonEmptyScheduleDays = (slots?: Slots) => {
-  if (typeof slots === "undefined") return [];
+  if (typeof slots === "undefined") return null;
```

Comment:
If slots aren't given yet we can't compute slots, so return null to differentiate b/w empty array when no slots are actually available in a month.

This handles the case of the past month where we would get 0 slots in the month and thus an empty array.

Reply:
@hariombalhara Could this lead to some issues if all slots for the current month are filled? Using the state after this merge triggered this error on our self-hosted instance.

```
TypeError: Cannot read properties of null (reading 'filter')
    at BookerComponent (/var/task/apps/web/.next/server/chunks/35157.js:313:150)
```

When navigating to a booking page that doesn't have any spots left for the current month (i.e. next two days).

Changing this back fixed the issue, but I'm not sure if it leads to different issues elsewhere.

Reply:
@dmonn yeah this is causing type error